### DISCUSSION TITLE
Using NMessageProvider and useMessage in Astro setup

### DISCUSSION THREAD

**Discussion by moseskamau338** [tusen-ai/naive-ui#5228] [Q&A]
Hi friends, am really greatful for this amazing library!

In a recent Astro project, I added Vue3 support and Tailwind then installed Naive UI. Every thing worked well so far untill I wanted to use forms and realized I have issues with `NMessageProvider`
 Am getting this error:
 
 ![Imgur](https://i.imgur.com/WyMne3J.png)

Here is my layout file

```html
<!DOCTYPE html>
<html lang="en">
    <head>...</head>
  <body>
    <LayoutWrapper>
         <slot />
    </LayoutWrapper>
  </body>
</html>
```

Then in the `<LayoutWrapper>` component:

```html
<script setup lang="ts">
import {NMessageProvider} from 'naive-ui'
</script>

<template>
    <NMessageProvider>
        <slot />
    </NMessageProvider>
</template>
```

Am then using the `useMessage()` composable in a `ContactForm.vue` file inside a `.astro` file

```html
<template>
// example taken form the docs on forms
</template>

<script lang="ts">
//@ts-ignore
import { FormInst, FormItemRule, useMessage } from 'naive-ui'
//@ts-ignore
import {ref} from 'vue'

export default{
  setup () {
    const formRef = ref<FormInst | null>(null)
    const message = useMessage()
// the rest of the code here

}
</script>
```

I know am doing this wrong...but what should I do? how should I approach this? 
Your thoughts are highly appreciated

**Comment by jahnli** [comment]
<NConfigProvider
  >
    <NLoadingBarProvider>
      <NModalProvider>
        <NDialogProvider>
          <NNotificationProvider>
            <NMessageProvider>
              <App />
            </NMessageProvider>
          </NNotificationProvider>
        </NDialogProvider>
      </NModalProvider>
    </NLoadingBarProvider>
  </NConfigProvider>
